src/libsyntax/errors/emitter.rs

```diff
@@ -23,6 +22,23 @@ use std::io;
 use std::rc::Rc;
 use term;
 
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Term(term::Error),
+}
+
+impl From<io::Error> for Error {
+    fn from(err: io::Error) -> Self {
+        Error::Io(err)
+    }
+}
+
+impl From<term::Error> for Error {
+    fn from(err: term::Error) -> Self {
+        Error::Term(err)
+    }
+}
```

Comment:
In terms of integration with libsyntax/libtest, however, it seems pretty onerous to always require a shim like this (or to merge *any* two libraries together)